nipst/nipst.go

```diff
@@ -228,12 +228,22 @@ func (nb *NIPSTBuilder) IsPostInitialized() bool {
 	return nb.postProver.Initialized()
 }
 
+func (nb *NIPSTBuilder) GetDataDirPath() string {
+	var d string
+	nb.cfgM.RLock()
```

Comment:
why do you need this lock? Where is it modified (after first initialization)?

Reply:
at initialization, this does not happen in a certain manner and potentially can happen in the same time, hence the lock